azure-stack-go-sdk/azure-stack-go-sdk-profiles.md

```diff
@@ -0,0 +1,302 @@
+
+1. Install Azure Go SDK and its dependencies (see previous section)
+2. Get metadata information from resource manager endpoint. Resource manager metadata endpoint returns a JSON file with required information to run Go samples. To retreive the metadata, open a web browser in your Azure Stack environment and use the url: `ResourceManagerURL`/metadata/endpoints?api-version=1.0
+
+The `ResourceManagerURL` varies based on the region name, machine name and external fully qualified domain name (FQDN) of your Azure Stack deployment: 
+
+|Environment|ResourceManagerURL|
+|-----------|------------------|
+|Development kit|https://management.local.azurestack.external/|
+|Integrated systems|https://management.(region).ext-(machine-name).(FQDN)|
+    
+Sample JSON file from a development kit environment (https://management.local.azurestack.external/metadata/endpoints?api-version=1.0): 
+
+```json
+    {
+        "galleryEndpoint": "https://portal.local.azurestack.external:30015/",
+        "graphEndpoint": "https://graph.windows.net/",
+        "portalEndpoint": "https://portal.local.azurestack.external/",
+        "authentication": {
+            "loginEndpoint": "https://login.windows.net/",
+            "audiences": [
+                "https://management.azurestackiXX.onmicrosoft.com/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+            ]
+        }
+    }
+```
+3. If not available, create a subscription and save the subscription ID to be used later. Instructions to create a subscription are [here](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-subscribe-plan-provision-vm)
+4. Create a service principal with "Subscription" scope and "Owner" role. Save the service principal's ID and secret. Instructions to create a service principal for Azure Stack are [here](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals)
+5. Now your Azure Stack environment is setup. In your code, import a service module from Go SDK profile. The current version of Azure Stack profile is "2017-03-09". For example, to import network module from "2017-03-09" profile:
+
+    ```go
+    package main
+
+    import "github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/network/mgmt/network"
+    ```
+
+6. In your function, create and authenticate a client with a New*Client function call. For example to create a virtual network client:
+
+    ```go
+    package main
+
+    import (
+        "github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/network/mgmt/network"
+        "github.com/Azure/go-autorest/autorest"  
+    ) 
+
+    func main() {
+        vnetClient := network.NewVirtualNetworksClientWithBaseURI("<baseURI>", "<subscriptionID>")
+        vnetClient.Authorizer = autorest.NewBearerAuthorizer(token)
+    }
+    ```
+
+    Set `baseURI` to the `ResourceManagerURL` value used in step 2.
+
+    Set `subscriptionID` to the subscription ID value from step 3.
+
+    To create token see Authentication section below
```

Comment:
Don't provide full code samples unless everything used in them is present. Instead present them as snippets consisting of the `import` statement(s) and the code which uses them only. Otherwise readers will cut-and-paste and end up with code that does not work.